File: src/main/java/com/rimlang/rim/syntax/MatchSyntaxNode.java
package com.rimlang.rim.syntax;

import com.rimlang.rim.lexer.Token;

import java.util.HashMap;
import java.util.List;

public class MatchSyntaxNode extends SyntaxNode {
    private GenericExpression genericExpression;
    private HashMap<List<Token>, List<SyntaxNode>> branches;

    public MatchSyntaxNode(GenericExpression genericExpression, HashMap<List<Token>, List<SyntaxNode>> branches) {
        this.genericExpression = genericExpression;
        this.branches = branches;
    }
    
}

File: src/main/java/com/rimlang/rim/syntax/node/control/MatchSyntaxNode.kt
package com.rimlang.rim.syntax.node.control

import com.rimlang.rim.lexer.token.Token
import com.rimlang.rim.syntax.node.expression.GenericExpression
import com.rimlang.rim.syntax.node.syntax.SyntaxNode

data class MatchSyntaxNode(
    val genericExpression: GenericExpression,
    val branches: HashMap<List<Token>, List<SyntaxNode>>,
    override val firstToken: Token
) : SyntaxNode(firstToken)